scheduler/clusterprocessor.go

```diff
@@ -78,10 +76,22 @@ func ClusterProcessor(commandFlag configuration.CommandFlag,
 
 		create := func() {
 			for _, cluster := range provider.Clusters {
-				err := runGCPCreate(cmc, ctx, cluster.Cluster)
+				clusterEndpoint, clusterCertCa, err := runGCPCreate(cmc, ctx, cluster.Cluster)
 				if err != nil {
 					color.Red(err.Error())
 				}
+
+				cluster.Cluster.Endpoint = clusterEndpoint
+				cluster.Cluster.CertCa = clusterCertCa
+
+				if err := platform.SetK8SContext(cluster.Cluster); err != nil {
+					color.Red(err.Error())
+				}
+
+				if err := vault.SetVaultConfiguration(opt, cluster.Cluster); err != nil {
```

Comment:
so this function is for google provider, but I guess we want to do this for any provider. So I think I would move this function a little bit down in this file, to the swtich command:

		switch commandFlag {
		case configuration.Create:
			create()
			if err := platform.SetK8SContext(cluster.Cluster); err != nil {
				color.Red(err.Error())
			}

			if err := vault.SetVaultConfiguration(opt, cluster.Cluster); err != nil {
				color.Red(err.Error())
			}
		case configuration.Apply:
			create()
			if err := platform.SetK8SContext(cluster.Cluster); err != nil {
				color.Red(err.Error())
			}

			if err := vault.SetVaultConfiguration(opt, cluster.Cluster); err != nil {
				color.Red(err.Error())
			}
		case configuration.Replace:
			delete()
			if err := platform.UnSetK8SContext(cluster.Cluster); err != nil {
				color.Red(err.Error())
			}

			if err := vault.UnSetVaultConfiguration(opt, cluster.Cluster); err != nil {
				color.Red(err.Error())
			}
			create()
			if err := platform.SetK8SContext(cluster.Cluster); err != nil {
				color.Red(err.Error())
			}

			if err := vault.SetVaultConfiguration(opt, cluster.Cluster); err != nil {
				color.Red(err.Error())
			}
		case configuration.Delete:
			delete()
			if err := platform.UnSetK8SContext(cluster.Cluster); err != nil {
				color.Red(err.Error())
			}

			if err := vault.UnSetVaultConfiguration(opt, cluster.Cluster); err != nil {
				color.Red(err.Error())
			}
		}